pgmpy/models/BayesianNetwork.py

```diff
@@ -585,7 +585,7 @@ def fit_update(self, data, n_prev_samples=None, n_jobs=1):
         self.add_cpds(*cpds)
         logger.disabled = False
 
-    def predict(self, data, stochastic=False, n_jobs=-1):
+    def predict(self, data, algo=None, stochastic=False, n_jobs=-1, **kwargs):
```

Comment:
The kwargs arguments provided here need to be passed to the Inference method initialization and the query calls later. Have a look at the various options in the query method for each of the inference and specify that in the docstring for the method.

Reply:
It does not look like there are keyword arguments for just the initialization of Inference methods. Is this not the case?

Reply:
@Nimish-4 That is correct. The query methods for some of the algorithms do accept additional arguments. I would suggest passing it to them.